src/openvic-simulation/map/Map.cpp

```diff
@@ -670,3 +670,82 @@ bool Map::generate_and_load_province_adjacencies(std::vector<LineObject> const&
 	);
 	return ret;
 }
+
+bool Map::load_climate_file(ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary([this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		if (identifier.empty()) {
+			Logger::error("Invalid climate identifier - empty!");
+			return false;
+		}
+
+		Climate* cur_climate = climates.get_item_by_identifier(identifier);
+		if (cur_climate == nullptr) {
+			ModifierValue values;
+			bool ret = modifier_manager.expect_modifier_value(move_variable_callback(values))(node);
+			ret &= climates.add_item({ identifier, std::move(values) });
+			return ret;
+		} else {
+			bool ret = expect_list([this, &identifier, &cur_climate](ast::NodeCPtr list_element) -> bool {
+				std::string_view cur_province_id;
+				bool ret = expect_identifier_or_string(assign_variable_callback(cur_province_id))(list_element);
+				Province* cur_province = provinces.get_item_by_identifier(cur_province_id);
+				if (cur_province != nullptr) {
+					cur_climate->add_province(cur_province);
+					return ret;
+				} else {
+					Logger::error("Invalid province id ", cur_province_id, " in definition of climate ", identifier, "!");
+					return false;
+				}
+			})(node);
+			return ret;
+		}
+	})(root);
+
+	lock_climates();
+
+	for (Climate const& cl : climates.get_items()) {
+		for(Province* prov : cl.get_provinces()) {
+			prov->climate = &cl;
+		}
+	}
+
+	return ret;
+}
+
+bool Map::load_continent_file(ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary([this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		if (identifier.empty()) {
+			Logger::error("Invalid continent identifier - empty!");
+			return false;
+		}
+
+		ModifierValue values;
+		ProvinceSetModifier::provinces_t prov;
+		bool ret = modifier_manager.expect_modifier_value_and_keys(move_variable_callback(values),
+			"provinces", ONE_EXACTLY, expect_list([this, &identifier, &prov](ast::NodeCPtr list_element) -> bool {
+				std::string_view cur_province_id;
+				bool ret = expect_identifier_or_string(assign_variable_callback(cur_province_id))(list_element);
+				Province* cur_province = provinces.get_item_by_identifier(cur_province_id);
+				if (cur_province != nullptr) {
+					prov.emplace_back(cur_province);
+					return ret;
+				} else {
+					Logger::error("Invalid province id ", cur_province_id, " in definition of continent ", identifier, "!");
+					return false;
+				}
+			})
+		)(node);
+		ret &= continents.add_item({ identifier, std::move(values), std::move(prov) });
+		return ret;
+	})(root);
+
+	lock_continents();
+
+	for (Continent const& con : continents.get_items()) {
+		for(Province* prov : con.get_provinces()) {
+			prov->continent = &con;
+		}
+	}
```

Comment:
With my changes in #109, this can be changed to:
```suggestion
	for (Continent const& continent : continents.get_items()) {
		if (continent.empty()) {
			Logger::warning("Continent ", continent.get_identifier(), " has no provinces!");
		} else {
			for(Province const* prov : continent.get_provinces()) {
				remove_province_const(prov)->continent = &continent;
			}
		}
	}
```